Integration tests using `stack build --profile` failing on macOS only

Following 2f192abbb5fef6da9127065e2d4abcdbe6e653ef, which is a simple change to unrelated parts of Stack, the integration tests are failing on macOS only. The tests affected are `mutable-deps` and `proper-rebuilds` and what they have in common is:

~~~haskell
stackCheckStderr ["build" , "--profile"] $ expectRecompilation mutablePackages
~~~

~~~haskell
stackCheckStderr ["build" , "--profile"] expectRecompilation
~~~

and failing with errors that refer to not finding modules and `Perhaps you haven't installed the profiling libraries for package ‘base-4.16.3.0’?` or `Perhaps you haven't installed the profiling libraries for package ‘unix-2.7.2.2’?`.

~~~text
2022-10-07T10:07:12.4256830Z filepath   > Building library for filepath-1.4.2.2..
2022-10-07T10:07:12.4257240Z filepath   > [1 of 3] Compiling System.FilePath.Posix
2022-10-07T10:07:12.4257530Z filepath   > 
2022-10-07T10:07:12.4258150Z filepath   > /Users/runner/work/stack/stack/test/integration/tests/mutable-deps/files/filepath-1.4.2.2/System/FilePath/Posix.hs:64:8: error:
2022-10-07T10:07:12.4258720Z filepath   >     Could not find module ‘Prelude’
2022-10-07T10:07:12.4259290Z filepath   >     Perhaps you haven't installed the profiling libraries for package ‘base-4.16.3.0’?
2022-10-07T10:07:12.4259880Z filepath   >     Use -v (or `:set -v` in ghci) to see a list of the files searched for.
2022-10-07T10:07:12.4260200Z filepath   >    |
2022-10-07T10:07:12.4260480Z filepath   > 64 | module System.FilePath.Posix
2022-10-07T10:07:12.4260770Z filepath   >    |        ^^^^^^^^^^^^^^^^^^^^^
2022-10-07T10:07:12.4261330Z filepath   >
~~~

In the case of `mutable-deps`:

* `filepath-1.4.2.2`/`System.FilePath.Posix` complains it cannot find `Prelude`, `Data.Char`, `Data.Maybe`, `Data.List`, `System.Environment`.

* `unix-compat-0.5.4`/`System.PosixCompat.Files` complains it cannot find `Prelude` or `System.Posix.Files` (`unix-2.7.2.2`).

In the case of `proper-rebuilds`:

* `files-1.0.0`/`Lib` complains it cannot find `Prelude`.

I have not been able (yet) to reproduce the problem locally.

--- a/.github/workflows/integration-tests.yml
+++ b/.github/workflows/integration-tests.yml
@@ -80,6 +80,12 @@ jobs:
         EOF
         fi
 
+        # In case GHCup hooks have been created, remove them
+        if [ -d $(stack path --stack-root)/hooks ]
+        then
+            rm -Rf $(stack path --stack-root)/hooks
+        fi
+
         # Do this in the same step as installing deps to get relevant env var modifications
         stack etc/scripts/release.hs check ${{ matrix.release-args }}
 

